src/Dotmailer.php

```diff
@@ -187,6 +188,22 @@ public function getContactAddressBooks(Contact $contact): array
         return $addressBooks;
     }
 
+    /**
+     * @param DataField $dataField
+     */
+    public function createContactDataField(DataField $dataField)
+    {
+        $this->response = $this->adapter->post('/v2/data-fields', $dataField->asArray());
+    }
+
+    /**
+     * @param DataField $dataField
+     */
+    public function deleteContactDataField(DataField $dataField)
+    {
+        $this->response = $this->adapter->delete('/v2/data-fields/' . $dataField->getName());
```

Comment:
What if the field is in use?

Reply:
Then that data is lost. Definitely something to be aware of and cautious of (as ever with a delete), but it mirrors the behaviour found in the Dotmailer UI